File: src/src_platform_swing/fmg/swing/utils/ProjSettings.java
package fmg.swing.utils;

import javax.swing.UIDefaults;
import javax.swing.UIManager;

import fmg.common.AProjSettings;
import fmg.common.ui.UiInvoker;
import fmg.core.mosaic.MosaicDrawModel;
import fmg.swing.img.Animator;

public final class ProjSettings extends AProjSettings {

    private ProjSettings() {}

    static {
        UiInvoker.DEFERRED = javax.swing.SwingUtilities::invokeLater;
        UiInvoker.ANIMATOR = Animator::getSingleton;
        UiInvoker.TIMER_CREATOR = Timer::new;


        try {
            UIDefaults uiDef = UIManager.getDefaults();
            java.awt.Color clr = uiDef.getColor("Panel.background");
            if (clr != null)
                MosaicDrawModel.DefaultBkColor = Cast.toColor(clr);
        } catch (Exception ex) {
            ex.printStackTrace(System.err);
        }
    }

    public static void init() {
        // implicit call static block
    }

}

File: FastMines_uwp/src/fmg/uwp/app/ProjSettings.cs
using System;
using Windows.UI;
using Windows.UI.Core;
using Windows.UI.ViewManagement;
using Fmg.Common.UI;
using Fmg.Core.Mosaic;
using Fmg.Uwp.Img;
using Fmg.Uwp.Utils;

namespace Fmg.Uwp.App {

    public class ProjSettings : Fmg.Core.App.AProjSettings {

        /// <summary> Mobile (true) or Desktop (false) </summary>
        public static bool IsMobile { get; }
        public static double MinTouchSize => Cast.DpToPx(48); // android recommended size

        static ProjSettings() {
            UiInvoker.Deferred = doRun => AsyncRunner.InvokeFromUi(() => doRun(), CoreDispatcherPriority.Normal);
            UiInvoker.Animator = () => Animator.Singleton;
            UiInvoker.TimerCreator = () => new Timer();

            try {
                var uiSettings = new UISettings();

                Color clrCell;
                Color clrBk;
                UIElementType forCell = UIElementType.ButtonFace; // UIElementType.Window
                UIElementType forBkrd = UIElementType.PageBackground;
                try {
                    // desktop
                    clrCell = uiSettings.UIElementColor(forCell);
                    clrBk   = uiSettings.UIElementColor(forBkrd);
                    IsMobile = false;
                } catch (ArgumentException) {
                    IsMobile = true;
                    try {
                        // mobile
                        const int magic = 1000;
                        clrCell = uiSettings.UIElementColor(magic + forCell);
                        clrCell = uiSettings.UIElementColor(magic + forBkrd);
                    }
                    catch (Exception) {
                        // wtf??
                        clrCell = MosaicDrawModelConst.DefaultCellColor.ToWinColor();
                        clrCell = MosaicDrawModelConst.DefaultBkColor  .ToWinColor();
                    }
                }
                MosaicDrawModelConst.DefaultCellColor = clrCell.ToFmColor();
                MosaicDrawModelConst.DefaultBkColor   = clrBk  .ToFmColor();
            } catch (Exception ex) {
                System.Diagnostics.Debug.WriteLine(ex.Message);
                System.Diagnostics.Debug.Assert(false, ex.Message);
            }
        }

        public static void Init() {
            // implicit call static block
        }

    }

}
